telegram/chatmember.py

```diff
@@ -29,7 +29,15 @@
 
 
 class ChatMember(TelegramObject):
-    """This object contains information about one member of a chat.
+    """Base class for Telegram ChatMember Objects..
+    Currently, the following 6 types of chat members are supported:
+
+    * :class:`telegram.ChatMemberOwner`
+    * :class:`telegram.ChatMemberAdministrator`
+    * :class:`telegram.ChatMemberMember`
+    * :class:`telegram.ChatMemberRestricted`
+    * :class:`telegram.ChatMemberLeft`
+    * :class:`telegram.ChatMemberBanned`
```

Comment:
did we want to deprecated the arguments/attributes that are not common to all the subclasses?

Reply:
Only `status` and `user` are common in all subclasses, so you mean depreciate all other attributes from base class? If yes then how do we depreciate ig only way is by overriding `__getattr__()` and checking if user is trying to acess some other attribute than (`user`, `status`) then show the warning that these attributes are deprecated, and will be removed in future 🤔

Reply:
Mh, I guess I'd also be fine with just stating it in the docs …

Reply:
And if we override `__getattr__()` doesn't it'll apply to it's subclasses too..

Reply:
> Mh, I guess I'd also be fine with just stating it in the docs …

Hmmm okay, that would be better imo